### PR TITLE
[10] Connection section

### LINKED ISSUE #10: Add section about working with Connections to Docs
Add section about working with Connections to Docs

### PR DESCRIPTION
**Pull request by mcdoker18**
This closes #10

### CODE REVIEW

File: source/ref_connections.rst [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+We have been testing the following Docker repositories:
+
+    * `Docker Hub <https://docs.docker.com/docker-hub/>`_
+    * `Nexus <https://help.sonatype.com/repomanager3/formats/docker-registry>`_
+    * `Google Container Registry <https://cloud.google.com/container-registry/docs/>`_
+    * `Azure Container Registry <https://docs.microsoft.com/en-in/azure/container-registry/container-registry-intro>`_
+
+.. warning::
+    Every docker registry has its authorization specificity.
+    But you must be able to authorize by a username and password. Read documentation.
```
Comment (vlad-tokarev) [inline source/ref_connections.rst]: Read the documentation

**Review by vlad-tokarev — APPROVED** [review]
Looks cool

File: source/ref_connections.rst [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+*********************
+Connection management
+*********************
+
+Connections can be managed using the following ways.
+
+Swagger UI
+----------
+
+Swagger UI is available at http:://api-service/swagger/index.html URL.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
Swagger UI is available at http://api-service/swagger/index.html URL.
```

File: source/ref_connections.rst [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    * :ref:`ref_connections:Google Cloud Storage`
+    * :ref:`ref_connections:Azure Blob storage`
+    * :ref:`ref_connections:GIT`
+    * :ref:`ref_connections:Docker`
+    * :ref:`ref_connections:Amazon Elastic Container Registry`
+
+S3
+--
+
+A S3 connection allows interacts with `s3 API <https://docs.aws.amazon.com/en_us/AmazonS3/latest/dev/Welcome.html>`_.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
An S3 connection allows interactions with `s3 API <https://docs.aws.amazon.com/en_us/AmazonS3/latest/dev/Welcome.html>`_.
```

File: source/ref_connections.rst (hunk 1 above)
Comment (mcdoker18) [reply source/ref_connections.rst]: Fixed

File: source/ref_connections.rst [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+------------------
+
+Odahu-flow uses the `Blob storage <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-blobs-introduction>`_ in Azure to store:
+
+    * model trained artifacts.
+    * input data for ML models.
+
+Before usage, make sure that:
+
+    * You have created a storage account . `Create a storage account <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-storage-account>`_.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
    * You have created a storage account . `Create a storage account <https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-storage-account>`_.
```

File: source/ref_connections.rst [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+    * You have created a storage container in the storage account . `Create a container <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-container>`_.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
    * You have created a storage container in the storage account . `Create a container <https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-container>`_.
```

File: source/ref_connections.rst [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+    * You have created an SAS token. `Create an account SAS <https://docs.microsoft.com/ru-ru/rest/api/storageservices/create-account-sas>`_.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
    * You have created a SAS token. `Create an account SAS <https://docs.microsoft.com/en-us/rest/api/storageservices/create-account-sas>`_.
```

File: source/ref_connections.rst [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+    * `Google Container Registry <https://cloud.google.com/container-registry/docs/>`_
+    * `Azure Container Registry <https://docs.microsoft.com/en-in/azure/container-registry/container-registry-intro>`_
+
+.. warning::
+    Every docker registry has its authorization specificity.
+    But you must be able to authorize by a username and password. Read the documentation.
+
+Before usage, make sure that:
+
+    * You have had a username and password.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
    * You have a username and password.
```

File: source/ref_connections.rst [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+
+The following fields of connection API are required:
+
+    * ``spec.type`` - It must be equal **docker**.
+    * ``spec.username`` - docker registry username.
+    * ``spec.password`` - docker registry password.
+    * ``spec.uri`` - docker registry host.
+
+.. warning::
+    Connection URI must not contains an URI schema.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
    Connection URI must not contain an URI schema.
```

File: source/ref_connections.rst [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+`Amazon Elastic Container Registry <https://docs.aws.amazon.com/AmazonECR/latest/userguide/what-is-ecr.html>`_  is a managed AWS Docker registry.
+This type of connection is used for pulling and pushing of the Odahu packager result Docker images.
+
+.. note::
+    The Amazon Docker registry does not support a long-lived credential and requires explicitly to create a repository for every image.
+    These are the reasons why we create a dedicated type of connection for the ECR.
+
+Before usage, make sure that:
+
+    * You have created an ECR repository. `Creating a ECR Repository <https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-create.html>`_.
```
Comment (maxkochubey) [inline source/ref_connections.rst]:
```suggestion
    * You have created an ECR repository. `Creating an ECR Repository <https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-create.html>`_.
```

**Review by maxkochubey — CHANGES_REQUESTED** [review]

**Review by maxkochubey — APPROVED** [review]

File: source/ref_connections.rst [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+    * You have created a storage account . `Create a storage account <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-storage-account>`_.
+    * You have created a storage container in the storage account . `Create a container <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-container>`_.
+    * You have created an SAS token. `Create an account SAS <https://docs.microsoft.com/ru-ru/rest/api/storageservices/create-account-sas>`_.
+
+The following fields of connection API are required:
+
+    * ``spec.type`` - It must be equal **azureblob**.
+    * ``spec.keySecret`` - Odahu-flow uses the `shared access signatures <https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview>`_ to authorize in Azure.
+      The key has the following format: "<primary_blob_endpoint>/<sas_token>".
+    * ``spec.uri`` -  GCS compatible URI, for example <bucket-name>/dir1/dir2/
```
Comment (karbyshevds) [inline source/ref_connections.rst]: * ``spec.uri`` -  Azure storage compatible URI, for example <bucket-name>/dir1/dir2/

**Review by karbyshevds — CHANGES_REQUESTED** [review]

**Pull request merged**